src/main/java/dal/fatih/todoly/Main.java

```diff
@@ -55,28 +55,54 @@ public static void listAllTasks() {
             }
         }
     }
+
     public static void showTaskDetails() {
         System.out.print("Task Id :");
         String taskId = scn.nextLine();
-        Task task=tasks.get(taskId);
+        Task task = tasks.get(taskId);
         if (task != null) {
             System.out.println(tasks.get(taskId));
         } else {
             System.out.println("Task not found");
         }
     }
-    public static void deleteTask(){
+
+    public static void deleteTask() {
         System.out.print("Task Id :");
         String taskId = scn.nextLine();
-        Task task=tasks.get(taskId);
+        Task task = tasks.get(taskId);
         if (task != null) {
             tasks.remove(taskId);
-            System.out.println(task.getTitle() +"titled task deleted");
+            System.out.println(task.getTitle() + "titled task deleted");
         } else {
             System.out.println("Task not found");
         }
     }
-    
+
+    public static void filterTask() {
+
+        try {
+            DateFormat dueDateParser = new SimpleDateFormat("dd/MM/yyyy");
+            ArrayList<Task> taskList = new ArrayList<>(tasks.values());
+            System.out.println("Last Date");
+            String lastDateInput = scn.nextLine();
+            Date lastDate = dueDateParser.parse(lastDateInput);
+
+            int none = 0;
+            for (Task task : taskList) {
+                if (task.getDate().before(lastDate)) {
+                    System.out.println(task);
+                    none++;

```

Comment:
It is not very easy to understand what this is for. Instead of printing the found tasks right away, you can collect them in another list and then you can check the size if it is zero instead to do what you are trying to do with this `none` variable. 

Don't use an additional variable to do that to sum it.